src/commands/deploy.js

```diff
@@ -1,131 +1,18 @@
-		if (!envVariables[k]) {
-			throw new Error(`env var ${k} not defined - aborting`);
-		}
-	});
-
 module.exports = {
 	command: 'deploy',
-	description: 'deploy the current application to production',
+	describe: 'deploy the current application to production',
 	builder: yargs =>
-		yargs.options({
-			version: {
-				alias: 'v',
-				default: CI_BUILD_NUMBER,
-				demandOption: true,
-				describe: 'The version ID to deploy',
-			},
-			servicesId: {
-				default: 'default',
-				describe: 'The GAE service name',
-			},
-			incrementWait: {
-				default: 60000, // 1 minute
-				describe: 'The delay between migration increments',
-			},
-			incrementPercentage: {
-				default: 50,
-				describe:
-					'The percentage of traffic to migrate in each increment',
-			},
-			maxInstances: {
-				default: 196,
-				describe:
-					'The maximum number of available instances in a GAE deployment',
-			},
-			deployCount: {
-				default: 1,
-				describe: 'The number of parallel versions to deploy',
-			},
-			targetTraffic: {
-				default: 100,
-				describe:
-					'The total amount of traffic to be migrated to the deployment',
-			},
-			pollWait: {
-				default: 10000, // 10 seconds
-				describe: 'The time to wait between progress checks',
-			},
-			noCanary: {
-				describe: 'Disable canary test',
-			},
-		}),
-	handler: argv => {
-		const envVariables = {
-			API_HOST: 'api.meetup.com',
-			COOKIE_ENCRYPT_SECRET,
-			CSRF_SECRET,
-			DEV_SERVER_PORT: '8080',
-			NEW_RELIC_APP_NAME,
-			NEW_RELIC_LICENSE_KEY,
-			PHOTO_SCALER_SALT,
-		};
-		validateEnv(envVariables);
-
-		const versionIds = Array.apply(null, {
-			length: argv.deployCount,
-		}).map((_, i) => `${argv.version}-${i}`);
-
-		return cloudApi.auth().then(({ auth, appsId }) => {
-			const indent = chalk.yellow(' >');
-			const config = Object.assign(
-				{ envVariables, versionIds, auth, appsId, indent },
-				argv
-			);
-			console.log(chalk.blue(`Deploying version ${config.version}`));
-
-			const { deploy, versions, migrate } = getApi(config);
-			return versions.validate
-				.sufficientQuota()
-				.then(deploy.create)
-				.then(versions.start)
-				.then(
-					() =>
-						config.noCanary ||
-						Promise.all(versionIds.map(runE2EWithRetry))
-				)
-				.catch(error => {
-					console.log(chalk.red(`Stopping deployment: ${error}`));
-					console.log(chalk.red('Cleaning up failed deployment...'));
-					// clean up deployed versions and exit
-					return deploy.del().then(() => process.exit(1));
-				})
-				.then(() =>
-					migrate().catch(error => {
-						console.log(chalk.red(`Migration stopped: ${error}`));
-						if (
-							!(error instanceof versions.validate.RedundantError)
-						) {
-							// no need to return non-zero exit code for newer deployment
-							process.exit(1);
-						}
-					})
-				);
-		});
-	},
+		yargs
+			.options({
+				servicesId: {
+					default: 'default',
+					describe: 'The GAE service name',
+				},
+				pollWait: {
```

Comment:
all `deploy` commands use these two options because they are required by the API middleware